bobweb/bob/database.py

```diff
@@ -142,6 +142,31 @@ def update_user_in_db(update: Update):
         increment_chat_member_message_count(update.effective_chat.id, update.effective_user.id)
 
 
+def get_least_recently_seen_proverb_for_chat(chat_id):
```

Comment:
```suggestion
def get_least_recently_seen_proverb_for_chat(chat_id):
    annotated_chatproverb: FilteredRelation = FilteredRelation('chatproverb', condition=Q(chatproverb__chat__id=chat_id))
    # Joins chatProverb with proverb. Gets least recently seen proverb or oldest new proverb that has not yet been
    # seen in the chat
    # Reult: Tuple[proverb.id, chatproverb.id, chatproverb.last_appeared]
    result = Proverb.objects.annotate(chat_proverb=annotated_chatproverb)\
        .values('id', 'chatproverb__id', 'chatproverb__last_appeared')\
        .order_by('chat_proverb__last_appeared', 'id')\
        .first()

    if result['chatproverb__id'] is None:
        # proverb has not yet appeared on chat
        chat_proverb = ChatProverb.objects.create(chat_id=chat_id, proverb_id=result['id'])
    else:
        chat_proverb = ChatProverb.objects.get(id=result['chatproverb__id'])

    chat_proverb.number_of_appearances += 1
    chat_proverb.last_appeared = datetime.now(tz=pytz.UTC)  # Saved as normalized UTC+-0 time
    chat_proverb.save()
    return chat_proverb.proverb
```

Tuon ekan haun vastaava sqlite:
```sql
select p.id as proberbId, cp.id as chatProverbId, cp.last_appeared
from bobapp_proverb p left outer join bobapp_chatproverb cp on p.id = cp.proverb_id
-- Order by last appearance of proverb in chat. In SQLite NULL value is always smallest. Use proverb id as secondary order by.
-- This returns either proverb from which appearance is the longest of time or oldest proverb added that
-- has not yet been posted to the chat
order by cp.last_appeared asc, p.id asc
```

Jäin tätä pohtimaan ja tarjoan tälläistä ratkaisua ongelmaan. Ei tekisi kuin max 3 queryä. Ekana hakee viimeisimmän Tuple[proverb.id, chatproverb.id, chatproverb.last_appeared], jonka last appeared on vanhin TAI jos löytyy kannasta sellainen proverb jolle ei ole chatissa vielä chatproverb riviä persistoitu, otetaan niistä vanhin proverb id:n perusteella.

Sit muuten logiikka sama kuin sun toteutuksessa, eli jos ei ole vielä chatproverb persistoitu, niin persistoidaan uusi. Muuten haetaan id:llä se kannasta. Päivitetään `number_of_appearances` ja `last_appeared`, jonka jälkeen palautetaan `proverb` olio